File: src/Poker/hand/Straight.java
package Poker.hand;

import Poker.game.DeckConstants.Value;

public class Straight extends Hand
{
   protected Value high;
   
   public Straight(Value high)
   {
      super(Rank.STRAIGHT);
      this.high = high;
   }
   
   public Value getHigh()
   {
      return high;
   }
   
   public String toString()
   {
      return "Straight - " + high.toString() + " high";
   }
   
   public int compare(Hand handToCompare)
   {
      int iCompare = compareRank(handToCompare);
      if (iCompare != 0)
      {
         return iCompare;
      }
      
      Value highToCompare = ((Straight) handToCompare).getHigh();
      return high.getIndex() - highToCompare.getIndex();
   }
}

File: server/hand/straight.js
let Hand = require('./hand');

let Straight = (highValue) =>
{
	const STRAIGHT = 4;
	
	let rank = Hand(STRAIGHT);
	
	let self =
	{
		compare: (hand) =>
		{
			let rankCompare = rank.compare(hand);
			if (rankCompare)
			{
				return rankCompare;
			}
			
			return self.compareSameRank(hand);
		},
		compareSameRank: (straight) =>
		{
			return highValue - straight.getHighValue();
		},
		getHighValue: () => highValue,
		toString: () =>
		{
			
		}
	}
	
	return self;
}

module.exports = Straight;
